### PR TITLE
Redirect an old png badge with a number as a color; test on [static]

### PR DESCRIPTION
**Pull request by platan**
Fixes https://github.com/badges/shields/issues/3260

Problem happens when a value of a color in an old PNG static badge is a number: http://localhost:8080/my-label/my-message.png?color=1. In this case `color` in `queryParams` is a number. 
https://github.com/badges/shields/blob/0a0b5b3f031178851c9d295b9967d81ede052723/core/server/server.js#L203-L212

Surprisingly service test listed below is passing currently on master - value `1` is represented in `queryParams` as a String (only in test). 
`services/static-badge/static-badge.tester.js`
```js
t.create('Old static badge with a number as a color')
  .get('/foo/bar.png?color=1', { followRedirect: false })
  .expectStatus(301)
  .expectHeader('Location', '/badge/foo-bar-1.png')
```

Moreover I added some code + description allowing to debug server.

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{
```
Comment (platan) [inline .vscode/launch.json]: This configuration comes from https://github.com/Microsoft/vscode-recipes/tree/master/nodemon#configure-vs-code-debugging-with-a-launchjson-file

File: core/badge-urls/make-badge-url.js [hunk 2]
```diff
@@ -62,7 +62,10 @@ function staticBadgeUrl({
   namedLogo,
   format = 'svg',
 }) {
-  const path = [label, message, color].map(encodeField).join('-')
+  const path = [label, message, color]
+    .map(s => s.toString())
```
Comment (paulmelnikow) [inline core/badge-urls/make-badge-url.js]: Could you move the coercion to the calling side instead? I feel like it's reasonable for this function to expect strings. It's really the web framework that is a bit unreasonable, about giving us different types depending on the input.

File: services/static-badge/static-badge.tester.js [hunk 3]
```diff
@@ -62,3 +62,8 @@ t.create('Old static badge')
   .get('/foo/bar.png?color=blue', { followRedirect: false })
   .expectStatus(301)
   .expectHeader('Location', '/badge/foo-bar-blue.png')
+
+t.create('Old static badge without a color')
```
Comment (paulmelnikow) [inline services/static-badge/static-badge.tester.js]: Could you add a test of the behavior that you're fixing?

File: core/badge-urls/make-badge-url.js (hunk 2 above)
Comment (platan) [reply core/badge-urls/make-badge-url.js]: Thanks! Done.

File: services/static-badge/static-badge.tester.js (hunk 3 above)
Comment (platan) [reply services/static-badge/static-badge.tester.js]: I can. But as I wrote in PR description, this test is passing without a fix:
```js
t.create('Old static badge with a number as a color')
  .get('/foo/bar.png?color=1', { followRedirect: false })
  .expectStatus(301)
  .expectHeader('Location', '/badge/foo-bar-1.png')
```
A `color` value (in `queryParams`) is a string, not a number.

Comment (platan) [reply services/static-badge/static-badge.tester.js]: In my opinion is it not worth to add a test which gives a false sense of working solution.

File: core/server/server.js [hunk 4]
```diff
@@ -207,7 +207,8 @@ module.exports = class Server {
       const redirectUrl = staticBadgeUrl({
         label,
         message,
-        color,
+        // Fixes https://github.com/badges/shields/issues/3260
+        color: color && color.toString(),
```
Comment (paulmelnikow) [inline core/server/server.js]:
```suggestion
        color: color ? color.toString() : undefined,
```

File: services/static-badge/static-badge.tester.js (hunk 3 above)
Comment (paulmelnikow) [reply services/static-badge/static-badge.tester.js]: Ahhh I understand now. For some reason when this request is made from the test, Scoutcamp is returning the query string param as a string, but when it's made to the real server it gets a number. That's really puzzling.

I'm hoping to replace Scoutcamp (#3328) so I don't feel like it's worth digging further into why it's happening.

**Comment by platan** [comment]
Thanks for merging it :-)

**Pull request merged**